.github/workflows/codechecker.yaml

```diff
@@ -0,0 +1,62 @@
+name: Run Static Analysis
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  RunCodechecker:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+
+      - name: Install Dependencies
+        run: |
+          curl -O https://efa-installer.amazonaws.com/aws-efa-installer-1.30.0.tar.gz
+          tar -xf aws-efa-installer-1.30.0.tar.gz && cd aws-efa-installer
+          sudo ./efa_installer.sh -y
+          # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start-nccl-base.html#nccl-start-base-drivers
+          sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+          sudo apt-key adv --fetch-keys http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64/7fa2af80.pub
+          wget -O /tmp/deeplearning.deb \
+              http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/x86_64/nvidia-machine-learning-repo-ubuntu2004_1.0.0-1_amd64.deb
+          sudo dpkg -i /tmp/deeplearning.deb
+          sudo wget -O /tmp/cuda.pin https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-ubuntu2004.pin
+          sudo mv /tmp/cuda.pin /etc/apt/preferences.d/cuda-repository-pin-600
+          sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub
+          sudo add-apt-repository 'deb http://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/ /'
+          sudo apt-get update -y
+          sudo apt install nvidia-kernel-open-545 autoconf libhwloc-dev gcc-13 g++-13 cppcheck
+          sudo apt install -o Dpkg::Options::='--force-overwrite' libcudnn8-dev cuda-libraries-dev-12-3 cuda-toolkit-12-3 cuda-drivers-545
+          sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-13 13 --slave /usr/bin/g++ g++ /usr/bin/g++-13 --slave /usr/bin/gcov gcov /usr/bin/gcov-13
+          sudo update-alternatives --set gcc /usr/bin/gcc-13
+
+      - name: Run Configure
+        run: |
+          ./autogen.sh
+          ./configure --prefix=/opt/aws-ofi-nccl --with-mpi=/opt/amazon/openmpi --with-libfabric=/opt/amazon/efa --with-cuda=/usr/local/cuda --enable-platform-aws
+
+      - name: Run CodeChecker
+        uses: whisperity/codechecker-analysis-action@v1
+        id: codechecker
+        with:
+          build-command: "make"
+          ctu: true
```

Comment:
Have you had experience with codechecker before? Clang's static analyzer (scan-build) is more native to llvm/clang, so curious if this brings valuable additions. Got a sample artifact somewhere in your repo we could look at?

Reply:
> Clang's static analyzer (scan-build) is more native to llvm/clang, so curious if this brings valuable additions

Mainly the ability to run with CTU. See https://clang.llvm.org/docs/analyzer/user-docs/CrossTranslationUnit.html 

Quoting, emphasis mine:

> This process can be automated by other tools, like [CodeChecker](https://github.com/Ericsson/codechecker) and scan-build-py **(preference for the former).**

> **We actively develop CTU with CodeChecker** as the driver for [cross translation-unit analysis], scan-build-py is not actively developed for CTU. scan-build-py has various errors and issues, expect it to work only with the very basic projects only.

I don't think there's any strong reason to prefer a vanilla clang-static-analyzer job when this is running in CI and there's an off-the-shelf action we can use. It ultimately reads .cppcheck and .clang-tidy dotfiles to decide what it runs, so if scan-build learns to drive things better down the line, it's an easy swap back.

Reply:
Added a configuration file for this going forward.